### PR TITLE
update snakeyaml to 2.0

### PR DESCRIPTION
**Pull request by macfarla**
Update snakeyaml dependency

### CODE REVIEW

File: gradle/verification-metadata.xml [hunk 1]
```diff
@@ -5849,6 +5849,14 @@
...
          </artifact>
       </component>
+      <component group="org.yaml" name="snakeyaml" version="2.0">
+         <artifact name="snakeyaml-2.0.jar">
+            <sha256 value="880c9d896e4b74a06c549c15ca496450165d6909fa15d7e662bee8f6a66d7afa" origin="Generated by Gradle"/>
+         </artifact>
+         <artifact name="snakeyaml-2.0.pom">
+            <sha256 value="43c761f92b549c8b08fb992080253e49f0a983e544ef0663c1f4f9d68eb52616" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
```
Comment (siladu) [inline gradle/verification-metadata.xml:5859]: Can older versions of snake yaml be removed from here now? Not sure what the process is normally...thought it was automated though 🤷

**Review by gfukushima — APPROVED** [review]
lgtm

**Review by siladu — COMMENTED** [review]
This upgrade has breaking changes e.g. 
> Remove many deprecated methods (Backwards-incompatible)

Do we know that we're not going to see an error at runtime related to this or are we relying on automated tests?

**Comment by macfarla** [comment]
closed in favour of #5265

**Pull request closed without merging**